Add styles for Spinner component

mmtr opened the pull request:
Closes #8566.

It adds the needed styles by the `Spinner` component, so it is displayed correctly even if the WordPress Core styles are not available.

Note that the spinner provided by the WordPress Core styles uses an image. In order to don't handle any image during the build process, this component renders a spinner using only CSS that should look like the original image.

![aug-10-2018 13-44-35](https://user-images.githubusercontent.com/1233880/43956202-8aca8668-9ca3-11e8-8696-c4e59b57f402.gif)
(Left: Original Image - Right: New CSS only)

mmtr committed:
Add styles for Spinner component

mmtr committed:
Add styles for Spinner component

mmtr commented:
@youknowriad @gziolo

gziolo commented on packages/components/src/spinner/index.js:
```diff
@@ -1,3 +1 @@
-const spinner = <span className="spinner is-active" />;
-
-export default () => spinner;
+export default () => <span className="components-spinner" />;
```
It's some kind of micro-optimization recommended by Facebook.

gziolo commented:
@karmatosed @jasmussen - can you confirm if the proposed approach is expected? There are a few similar components which need some action to be reusable outside of WordPress.

jasmussen commented:
I think Riad's instinct is right — we are building new components for future JavaScript powered WordPress components, and they need to be able to not have to rely on WordPress core styles. Long term we migth even want to redesign them but for now even if they duplicate the CSS they should look the same as WordPress core unless we have a good reason, and if we have a good reason we should track it on https://core.trac.wordpress.org/ticket/44749

youknowriad commented on packages/components/src/spinner/style.scss:
```diff
@@ -0,0 +1,33 @@
+@keyframes rotation {
```
Should we rename this `components-spinner-rotation` or something like that to avoid ambiguity. Alternatively we can move this animation to `_animations` because it looks generic enough

mmtr replied:
Agreed! I've moved it to `_animations`.

youknowriad commented on packages/components/src/spinner/style.scss:
```diff
@@ -0,0 +1,33 @@
+
+.components-spinner {
+	display: inline-block;
+	background-color: #808080;
```
Should we use one of our gray variables?

mmtr replied:
I had the same doubt, but didn't know which one use as none of them matches the color used by the current spinner image.

Any suggestion?

youknowriad replied:
This one looks close $dark-gray-200 ?

jasmussen replied:
Ah yes that's a good point. Definitely use the SASS colors if you can, we already have an existing ticket to bring those to core: https://core.trac.wordpress.org/ticket/40633

mmtr replied:
It looks nice with that color:

![aug-10-2018 13-44-35](https://user-images.githubusercontent.com/1233880/43956202-8aca8668-9ca3-11e8-8696-c4e59b57f402.gif)
(Left: Original Image - Right: New CSS only)

youknowriad reviewed (approved):
I like that it's only CSS 👏 I left some small comments but looks good otherwise.

youknowriad commented:
looks like there's a stylelint issue :)

mmtr committed:
Improvements to the Spinner component.

- Rotation animation can be reused
- Background color set from variable
- Fix tests

mmtr committed:
Fixing stylelint issue

mmtr committed:
Fixing stylelint issue

youknowriad commented:
Sorry for introducing those conflicts, do you mind running a rebase, I'll merge right after? Thanks

youknowriad commented:
Travis is very unstable at the moment, it doesn't let me merge this PR :)

mmtr committed:
Trigger Travis

mmtr committed:
Trigger Travis
